src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Config/EventHubOptions.cs

```diff
@@ -252,6 +252,8 @@ internal EventProcessorHost GetEventProcessorHost(IConfiguration config, string
                     sb.EntityPath = null; // need to remove to use with EventProcessorHost
                 }
 
+                creds.EventHubEndpoint = sb.Endpoint.OriginalString;
```

Comment:
It looks like we do not use `_receiverCreds` properly as cache. When we add a new EH listener we [overwrite ](https://github.com/Azure/azure-functions-eventhubs-extension/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerAttributeBindingProvider.cs#L67) and then [use](https://github.com/Azure/azure-functions-eventhubs-extension/blob/dev/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Triggers/EventHubTriggerAttributeBindingProvider.cs#L70) new value.

So when you will have 2 EventHub triggers the approach will not  work and it will stick to the latest endpoint.
EventHub1, EventHubConnectionString1 (namespace1)
EventHub2, EventHubConnectionString2 (namespace2)

I think '_receiverCreds' should be removed/redesigned overall.